pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/MLTransactionMetadataStore.java

```diff
@@ -88,95 +89,110 @@ public MLTransactionMetadataStore(TransactionCoordinatorID tcID,
         this.abortedTransactionCount = new LongAdder();
         this.transactionTimeoutCount = new LongAdder();
         this.appendLogCount = new LongAdder();
+        this.recoverTracker = recoverTracker;
+    }
 
+    public CompletableFuture<TransactionMetadataStore> init() {
```

Comment:
should better `public CompletableFuture<TransactionMetadataStore> init(TransactionRecoverTracker recoverTracker)` because originally not a local variable of MLTransactionMetadataStore.java. It only use to tracker the MetadataStore recover, when recovery is complete it will be garbage collected. If it change to a a local variable, we should clear the recover the metadata from tracker, because it will not be garbage collected. So I suggest `public CompletableFuture<TransactionMetadataStore> init(TransactionRecoverTracker recoverTracker)`